### ISSUE TITLE
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'

### ISSUE THREAD

**Issue by roland3564** [Grum999/CompositionHelper#1]
Steps to reproduce:
1) Import script (on Krita 5.0.8)
2) Enable extensions automatically
3) Reboot

Current Results:
The program echoes
```
`TypeError
Python 3.10.6: /usr/bin/python3
Tue Aug 16 16:43:07 2022

A problem occurred in a Python script.  Here is the sequence of
function calls leading up to the error, in the order they occurred.

 /home/n1/.local/share/krita/pykrita/compositionhelper/compositionhelper.py in start(self=<compositionhelper.compositionhelper.CompositionHelper object>)
  117             return
  118 
  119         CHMainWindow(PLUGIN_MENU_ENTRY, PLUGIN_VERSION)
  120 
  121 if __PLUGIN_EXEC_FROM__ == 'SCRIPTER_PLUGIN':
global CHMainWindow = <class 'compositionhelper.ch.chmainwindow.CHMainWindow'>
global PLUGIN_MENU_ENTRY = 'Composition Helper'
global PLUGIN_VERSION = '1.1.1'

 /home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py in __init__(self=<compositionhelper.ch.chmainwindow.CHMainWindow object>, chName='Composition Helper', chVersion='1.1.1')
  134 
  135         # initialise window
  136         self.__initialise()
  137 
  138         self.__opened = True
self = <compositionhelper.ch.chmainwindow.CHMainWindow object>
self.__initialise undefined

 /home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py in __initialise(self=<compositionhelper.ch.chmainwindow.CHMainWindow object>)
  204         self.cbxHelpers.setIconSize(self.__iconSizeHelper)
  205         for helper in CHHelpersDef.HELPERS:
  206             self.cbxHelpers.addItem(self.__buildHelperIcon(helper), f" {CHHelpersDef.HELPERS[helper]['label']}", helper)
  207         self.cbxHelpers.setCurrentIndex(list(CHHelpersDef.HELPERS.keys()).index(self.__settings.option(CHSettingsKey.HELPER_LAST_USED.id())))
  208         self.cbxHelpers.currentIndexChanged.connect(updateHelper)
self = <compositionhelper.ch.chmainwindow.CHMainWindow object>
self.cbxHelpers = <PyQt5.QtWidgets.QComboBox object>
self.cbxHelpers.addItem = <built-in method addItem of QComboBox object>
self.__buildHelperIcon undefined
helper = 'goldrect'

 /home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py in __buildHelperIcon(self=<compositionhelper.ch.chmainwindow.CHMainWindow object>, helper='goldrect')
  278 
  279         pen = QPen(Qt.SolidLine)
  280         pen.setWidth(1.5)
  281         pen.setColor(self.__palette.color(QPalette.ButtonText))
  282 
pen = <PyQt5.QtGui.QPen object>
pen.setWidth = <built-in method setWidth of QPen object>
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'
    __cause__ = None
    __class__ = <class 'TypeError'>
    __context__ = None
    __delattr__ = <method-wrapper '__delattr__' of TypeError object>
    __dict__ = {}
    __dir__ = <built-in method __dir__ of TypeError object>
    __doc__ = 'Inappropriate argument type.'
    __eq__ = <method-wrapper '__eq__' of TypeError object>
    __format__ = <built-in method __format__ of TypeError object>
    __ge__ = <method-wrapper '__ge__' of TypeError object>
    __getattribute__ = <method-wrapper '__getattribute__' of TypeError object>
    __gt__ = <method-wrapper '__gt__' of TypeError object>
    __hash__ = <method-wrapper '__hash__' of TypeError object>
    __init__ = <method-wrapper '__init__' of TypeError object>
    __init_subclass__ = <built-in method __init_subclass__ of type object>
    __le__ = <method-wrapper '__le__' of TypeError object>
    __lt__ = <method-wrapper '__lt__' of TypeError object>
    __ne__ = <method-wrapper '__ne__' of TypeError object>
    __new__ = <built-in method __new__ of type object>
    __reduce__ = <built-in method __reduce__ of TypeError object>
    __reduce_ex__ = <built-in method __reduce_ex__ of TypeError object>
    __repr__ = <method-wrapper '__repr__' of TypeError object>
    __setattr__ = <method-wrapper '__setattr__' of TypeError object>
    __setstate__ = <built-in method __setstate__ of TypeError object>
    __sizeof__ = <built-in method __sizeof__ of TypeError object>
    __str__ = <method-wrapper '__str__' of TypeError object>
    __subclasshook__ = <built-in method __subclasshook__ of type object>
    __suppress_context__ = False
    __traceback__ = <traceback object>
    args = ("setWidth(self, int): argument 1 has unexpected type 'float'",)
    with_traceback = <built-in method with_traceback of TypeError object>

The above is a description of an error in a Python program.  Here is
the original traceback:

Traceback (most recent call last):
  File "/home/n1/.local/share/krita/pykrita/compositionhelper/compositionhelper.py", line 119, in start
    CHMainWindow(PLUGIN_MENU_ENTRY, PLUGIN_VERSION)
  File "/home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py", line 136, in __init__
    self.__initialise()
  File "/home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py", line 206, in __initialise
    self.cbxHelpers.addItem(self.__buildHelperIcon(helper), f" {CHHelpersDef.HELPERS[helper]['label']}", helper)
  File "/home/n1/.local/share/krita/pykrita/compositionhelper/ch/chmainwindow.py", line 280, in __buildHelperIcon
    pen.setWidth(1.5)
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'
`
```
When the script is used.

Expected Results:
The script renders the window correctly

**Comment by Grum999** [comment]
Hi @roland3564 

Thanks for feedback!

I see were is the problem, but the strange thing is, it doesn't happen on my side :-/ 
Anyway, I'll fix it

But I see you're not using official version of Krita

In log: `Python 3.10.6: /usr/bin/python3` from OS
Official version of Krita use `Python 3.8.1 (default, Jul 15 2022, 09:34:17)` from Appimage

Also, version of Qt/PyQt you use is not the same than one provided with official Krita Appimage and I can't assure that plugins are working with version of library that are not fully compatible with Krita

Before downloading fix version 1.1.2, can you test current plugin version with official Appimage to see if problem persist?

Thanks.

Grum999

**Comment by roland3564** [comment]
> Hi @roland3564
> 
> Thanks for feedback!
> 
> I see were is the problem, but the strange thing is, it doesn't happen on my side :-/ Anyway, I'll fix it
> 
> But I see you're not using official version of Krita
> 
> In log: `Python 3.10.6: /usr/bin/python3` from OS Official version of Krita use `Python 3.8.1 (default, Jul 15 2022, 09:34:17)` from Appimage
> 
> Also, version of Qt/PyQt you use is not the same than one provided with official Krita Appimage and I can't assure that plugins are working with version of library that are not fully compatible with Krita
> 
> Before downloading fix version 1.1.2, can you test current plugin version with official Appimage to see if problem persist?
> 
> Thanks.
> 
> Grum999

Ok, I tried with the flatpak version of Krita, The script now works fine, it is quite possibly an inconsistency with OpenSUSE's  binaries, thanks for all.

**Comment by Grum999** [comment]
Fixed with [v1.1.2](./../releases/tag/1.1.2)
